src/types/validator.rs

```diff
@@ -184,9 +183,10 @@ impl Validator for Info {
     /// public key.
     fn verify_signature(&self, sign_bytes: &[u8], signature: &[u8]) -> bool {
         if let Some(pk) = &self.pub_key.ed25519() {
-            let verifier = Ed25519Verifier::from(pk);
-            if let Ok(sig) = ed25519::Signature::from_bytes(signature) {
-                return verifier.verify(sign_bytes, &sig).is_ok();
+            if let Ok(pk) = ed25519_dalek::PublicKey::from_bytes(&pk.into_bytes()) {
```

Comment:
this ain't pretty, but it's the only way to introduce minimal change.

Alternatively, we could follow the tendermint where they removed the signatory:
https://github.com/informalsystems/tendermint-rs/pull/522/files

so our changset (only ed25519) would look like this:
https://pastebin.mkaczanowski.pro/?66f06aeefe8be98d#BYfuFZM9YcVjAHuKjMzQSFasGBMCbKEN4ifBkSaKWV4v

Reply:
cc: @ParthDesai @joe-bowman what seems a better option here?

Reply:
The fork of signatory was used to fix a dependency issue with substrate libraries.

If Signatory doesn't offer us much, I'm all for removing the dependency.

Reply:
Taken that ed25519 is already moved out of signatory then the secp256k1 would be the only usecase, which makes me feel that doing the same thing as tendermint (get rid of signatory) is probably the better choice.

This would be the diff that completely removes signatory from tendermint light client:
https://pastebin.mkaczanowski.pro/?3a769b28278fc426#7roK6SKmmNif7eVtSzrWG4ZJQn7SVnAdTM5Pzwmeuc41

^^ I could also use @ParthDesai input on that :)